### PR TITLE
Update script for generating idl files [15483]

### PR DESCRIPTION
**Pull request by richiware**
## Contributor Checklist
- [ ] Commit messages follow the project guidelines. 
- [ ] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [ ] Changes are ABI compatible. 
- [ ] Changes are API compatible. 
- [ ] Documentation builds and tests pass locally. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: utils/scripts/update_generated_code_from_idl.sh [hunk 1]
```diff
@@ -3,9 +3,19 @@
 files_to_exclude=(
     './include/fastrtps/types/*'
     './include/fastdds/statistics/types.idl'
```
Comment (JLBuenoLopez-eProsima) [inline utils/scripts/update_generated_code_from_idl.sh:5]: If we are adding the other Fast DDS-Gen options to the IDL files with exceptions I suggest adding the final one as well. In this case the option to be set is `-d` followed by the output directory `${SOURCE_DIR}/src/cpp/statistics/types`

File: utils/scripts/update_generated_code_from_idl.sh [hunk 2]
```diff
@@ -3,9 +3,19 @@
...
+    )
+
+files_needing_typeobject=(
+    './test/unittest/dynamic_types/idl/Basic.idl'
+    './test/unittest/dynamic_types/idl/new_features_4_2.idl'
+    './test/unittest/dynamic_types/idl/Test.idl'
```
Comment (JLBuenoLopez-eProsima) [inline utils/scripts/update_generated_code_from_idl.sh:11]: These other IDL files have also the TypeObject in their corresponding folder within the Fast DDS project. I am unsure if it is required or not. But I think that either we update the TypeObject or if the test does not require it, remove the unused files from the project.

* `test/unittest/xtypes/idl/Types.idl`
* `test/unittest/xtypes/idl/WideEnum.idl`
* `test/xtypes/idl/Types.idl`

File: utils/scripts/update_generated_code_from_idl.sh [hunk 3]
```diff
@@ -40,7 +50,13 @@ for idl_file in "${idl_files[@]}"; do
 
     cd "${idl_dir}"
 
-    fastddsgen -replace "$file_from_gen"
+    # Detect if needs type_object.
+    [[ ${files_needing_typeobject[*]} =~ $idl_file ]] && to_arg='-typeobject' || to_arg=''
+
+    # Detect if needs case sensitive.
+    [[ ${files_needing_case_sensitive[*]} =~ $idl_file ]] && to_arg='-cs' || to_arg=''
```
Comment (JLBuenoLopez-eProsima) [inline utils/scripts/update_generated_code_from_idl.sh:57]: This instruction is overwritting the previous one, so the TypeObject is not being regenerated for any of the listed IDLs.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: utils/scripts/update_generated_code_from_idl.sh (hunk 1 above)
Comment (richiware) [reply utils/scripts/update_generated_code_from_idl.sh]: Done in 67797b9

File: utils/scripts/update_generated_code_from_idl.sh (hunk 2 above)
Comment (richiware) [reply utils/scripts/update_generated_code_from_idl.sh:10]: Done in 67797b9

File: utils/scripts/update_generated_code_from_idl.sh (hunk 3 above)
Comment (richiware) [reply utils/scripts/update_generated_code_from_idl.sh]: Done in 67797b9

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Comment by richiware** [comment]
Insert a new bash script. Not need of testing.

**Pull request merged**